### ISSUE TITLE
DataTransformerMlp Shuffles Test Data

### ISSUE THREAD

**Issue by truebluejason** [keras-team/autokeras#648]
### Reproducing Steps
```
# Get any classification problem dataset
# Split data into train, val, and test
# OneHotEncode y train, val, and test

mlpModule = MlpModule(loss=classification_loss, metric=Accuracy, searcher_args={}, verbose=True)
data_transformer = DataTransformerMlp(x_train)
train_data = data_transformer.transform_train(x_train, y_train)
val_data = data_transformer.transform_test(x_val, y_val)
test_data = data_transformer.transform_test(x_test)
fit_args = {
    "n_output_node": y_encoder.n_classes,
    "input_shape": x_train.shape,
    "train_data": train_data,
    "val_data": val_data
}
mlpModule.fit(n_output_node=fit_args.get("n_output_node"),
    input_shape=fit_args.get("input_shape"),
    train_data=fit_args.get("train_data"),
    test_data=fit_args.get("val_data"),
    time_limit=24 * 60 * 60)
pred = mlpModule.predict(test_data)
ans = y_test

print(Accuracy.compute(pred, ans))
```

### Expected Behavior
If mlpModule.fit's best model had an accuracy of 96%, the prediction should have an accuracy around 96% as well. Unfortunately, prediction's accuracy is around 49% for the specific dataset I've used.

### Setup Details
Include the details about the versions of:
 - OS type and version: Ubuntu 18.04.1
 - Python: 3.6
 - autokeras: 0.4.0

**Comment by truebluejason** [comment]
Since I appear to have not have push access to the repository, here's my proposed solution:

File: `autokeras/backend/torch/data_transformer.py`
```
    def transform_train(self, data, targets=None, batch_size=None, shuffle=True):
        data = (data - self.mean) / self.std
        data = np.nan_to_num(data)
        dataset = self._transform([], data, targets)

        if batch_size is None:
            batch_size = Constant.MAX_BATCH_SIZE
        batch_size = min(len(data), batch_size)

        return DataLoader(dataset, batch_size=batch_size, shuffle=shuffle)

    def transform_test(self, data, target=None, batch_size=None):
        return self.transform_train(data, targets=target, batch_size=batch_size, shuffle=False)
```

Please let me know how you like it @jhfjhfj1. Thanks!
